In [31]:
from google.colab import drive
drive.mount('/content/gdrive')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [32]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn

In [33]:
df = pd.read_csv('/content/gdrive/My Drive/College_project/movie_metadata.csv')

In [34]:
df.head()

color      director_name  ...  aspect_ratio  movie_facebook_likes
0  Color      James Cameron  ...          1.78                 33000
1  Color     Gore Verbinski  ...          2.35                     0
2  Color         Sam Mendes  ...          2.35                 85000
3  Color  Christopher Nolan  ...          2.35                164000
4    NaN        Doug Walker  ...           NaN                     0

[5 rows x 28 columns]

In [35]:
df.columns

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')

For movie Recomendation we need only few columns like


1.   Director_name
2.   actor_1_name
3.   actor_2_name
4.   actor_3_name
5.   genrres
6.   movie_list







In [36]:
df = df.loc[:,['director_name','actor_1_name','actor_2_name','actor_3_name','genres','movie_title']]

In [37]:
df.head()

director_name  ...                                        movie_title
0      James Cameron  ...                                            Avatar 
1     Gore Verbinski  ...          Pirates of the Caribbean: At World's End 
2         Sam Mendes  ...                                           Spectre 
3  Christopher Nolan  ...                             The Dark Knight Rises 
4        Doug Walker  ...  Star Wars: Episode VII - The Force Awakens    ...

[5 rows x 6 columns]

In [38]:
df['director_name'].value_counts()

Steven Spielberg    26
Woody Allen         22
Clint Eastwood      20
Martin Scorsese     20
Ridley Scott        17
                    ..
Ingmar Bergman       1
Edward Hall          1
Michael Winner       1
Sara Newens          1
Kelly Makin          1
Name: director_name, Length: 2398, dtype: int64

In [39]:
df['director_name'].isna().sum()

104

In [40]:
df['actor_1_name'].isna().sum()

7

In [41]:
df['actor_2_name'].isna().sum()

13

In [42]:
df['actor_3_name'].isna().sum()

23

In [43]:
df['genres'].isna().sum()

0

In [44]:
df['movie_title'].isna().sum()

0

## Observation
We look that we have in the 'director_name' feature we have 104 Nan value

*   In the 'actor_1_name' feature we have 7 Nan value
*   In the 'actor_2_name' feature we have 13 Nan value
*   In the 'actor_3_name' feature we have 23 Nan value
*   In the 'genres' feature we have 0 Nan value
*   In the 'movie_title' feature we have 0 Nan value


In [45]:
df.head()

director_name  ...                                        movie_title
0      James Cameron  ...                                            Avatar 
1     Gore Verbinski  ...          Pirates of the Caribbean: At World's End 
2         Sam Mendes  ...                                           Spectre 
3  Christopher Nolan  ...                             The Dark Knight Rises 
4        Doug Walker  ...  Star Wars: Episode VII - The Force Awakens    ...

[5 rows x 6 columns]

In [46]:
df['genres'] = df['genres'].str.replace('|',' ')
df['genres'] = df['genres'].str.replace('Sci-Fi' , 'science friction')

In [47]:
df.head()

director_name  ...                                        movie_title
0      James Cameron  ...                                            Avatar 
1     Gore Verbinski  ...          Pirates of the Caribbean: At World's End 
2         Sam Mendes  ...                                           Spectre 
3  Christopher Nolan  ...                             The Dark Knight Rises 
4        Doug Walker  ...  Star Wars: Episode VII - The Force Awakens    ...

[5 rows x 6 columns]

In [48]:
df['actor_3_name'] = df['actor_3_name'].str.replace('-',' ')

In [49]:
df.head()

director_name  ...                                        movie_title
0      James Cameron  ...                                            Avatar 
1     Gore Verbinski  ...          Pirates of the Caribbean: At World's End 
2         Sam Mendes  ...                                           Spectre 
3  Christopher Nolan  ...                             The Dark Knight Rises 
4        Doug Walker  ...  Star Wars: Episode VII - The Force Awakens    ...

[5 rows x 6 columns]

In [52]:
df.to_csv('data_p1.csv',index=False)